source/server/http_test_server_filter.cc

```diff
@@ -79,6 +103,10 @@ HttpTestServerDecoderFilter::decodeTrailers(Envoy::Http::RequestTrailerMap&) {
 void HttpTestServerDecoderFilter::setDecoderFilterCallbacks(
     Envoy::Http::StreamDecoderFilterCallbacks& callbacks) {
   decoder_callbacks_ = &callbacks;
+  // TODO(oschaaf): this adds locking in the hot path. Consider moving this into a separate
+  // extension, which will also allow tracking multiple points via configuration.
+  last_request_delta_ns_ =
```

Comment:
Could this stopwatch call be conditional on `!base_config_.emit_previous_request_delta_in_response_header().empty()` since we don't need it otherwise?

Reply:
We can, but that would prevents this feature from working with per-request reconfiguration.

Reply:
Probably should keep it unconditional, since I don't know of any actual performance impact, and it's supposed to cost only 100ns to take a lock.